app/demo-loop/diagnostic/EnergyDiagnostic.hh

```diff
@@ -217,10 +218,13 @@ CELER_FUNCTION void EnergyDiagnosticLauncher<M>::operator()(ThreadId tid) const
         return;
     }
 
+    celeritas::LevelStateAccessor lsa(
+        &states_.geometry, tid, states_.geometry.level[tid]);
+
```

Comment:
Unfortunately this won't work because it has to have the same interface between ORANGE and VecGeom: this class was a major hack and you've exposed a terrible breakage of encapsulation. To continue the hack we may need an `#if CELERITAS_USE_VECGEOM`...

Reply:
So you are saying that because of VecGeom we can't store `level` on `OrangeStateData`?

Reply:
VecGeom doesn't have an `OrangeStateData` ... it has a `VecGeomStateData`. This only worked because I'd defined `pos` as an array of Real3 with one per volume ID. As I said, it's a super-hack. So you can add the original code back inside a `#if CELERITAS_USE_VECGEOM`, and leave the new code in the `#else`. We will get this cleaned up eventually 😓

Reply:
Ah sorry, I see know that this comment on EnergyDiagnostic, i.e. not within ORANGE. Easy fix.